mesonbuild/cmake/interpreter.py

```diff
@@ -819,6 +827,10 @@ def configure(self, extra_cmake_options: T.List[str]) -> CMakeExecutor:
         cmake_args += cmake_get_generator_args(self.env)
         cmake_args += [f'-DCMAKE_INSTALL_PREFIX={self.install_prefix}']
         cmake_args += extra_cmake_options
+        if not any(arg.startswith('-DCMAKE_BUILD_TYPE=') for arg in cmake_args):
```

Comment:
I'm not sure that this is actually sufficient. CMake allows a space after the `-D`, so `-D        CMAKE_BUILD_TYPE=...` is equally valid. I'm not sure if it's sufficient to do `'CMAKE_BUILD_TYPE= in arg` or if we need to step up to `re.search(r'-D\s*CMAKE_BUILD_TYPE=', arg)`.

I'd also like to have a comment here that we are ignoring the `CMAKE_BUILD_TYPE` environment variable. I think it's fine to do so, but we should note that we're doing that.

Reply:
Doesn't the `cmake_defines_to_args` function ensure that there will not be a space following `-D`?

Reply:
Okay, yeah. I think this is fine then.